homeassistant/components/picnic/services.py

```diff
@@ -0,0 +1,175 @@
+"""Services for the Picnic integration."""
+
+from python_picnic_api import PicnicAPI
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import device_registry
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ATTR_AMOUNT,
+    ATTR_DEVICE_ID,
+    ATTR_PRODUCT_ID,
+    ATTR_PRODUCT_NAME,
+    CONF_API,
+    DOMAIN,
+    PRODUCT_DECORATOR_LABEL,
+    PRODUCT_DECORATOR_MAPPING,
+    PRODUCT_DECORATOR_PRICE,
+    PRODUCT_DECORATOR_VALIDITY,
+    SERVICE_ADD_PRODUCT_TO_CART,
+    SERVICE_SEARCH_PRODUCT,
+)
+
+
+class PicnicServiceException(Exception):
+    """Exception for Picnic services."""
+
+
+async def async_register_services(hass: HomeAssistant) -> None:
+    """Register services for the Picnic integration, if not registered yet."""
+
+    # Register the add_product service
+    if not hass.services.has_service(DOMAIN, SERVICE_ADD_PRODUCT_TO_CART):
+
+        async def async_add_product_service(call: ServiceCall):
+            # Get the picnic API client and call the handler with api client and hass
+            api_client: PicnicAPI = await get_api_client(
+                hass, call.data.get("device_id")
+            )
+            await handle_add_product(hass, api_client, call)
+
+        hass.services.async_register(
+            DOMAIN,
+            SERVICE_ADD_PRODUCT_TO_CART,
+            async_add_product_service,
+            schema=vol.Schema(
+                {
+                    vol.Optional(ATTR_DEVICE_ID): cv.string,
+                    vol.Optional(ATTR_PRODUCT_ID): cv.positive_int,
+                    vol.Optional(ATTR_PRODUCT_NAME): cv.string,
+                    vol.Optional(ATTR_AMOUNT): cv.positive_int,
```

Comment:
Accounting to the `service.yaml` file, this is required with a default of 1, however, according to the schema its completely optional?

Reply:
It shouldn't be required, I'll change the `service.yaml` definition.